src/composables/websocket.ts

```diff
@@ -0,0 +1,192 @@
+
+    const message: IMiddlewareWebSocketSubscriptionMessage = {
+      op: 'Subscribe',
+      payload: channelName,
+    };
+
+    wsClient.send(JSON.stringify(message));
+    channelsSubscribed[channelName] = true;
+  }
+
+  /**
+   * Handle websocket message event.
+   * @param {MessageEvent} message - The message event object.
+   */
+  function handleWebsocketMessage(message: MessageEvent) {
+    if (!message.data) {
+      return;
+    }
+    try {
+      const data = camelcaseKeysDeep(JSON.parse(message.data));
+      if (
+        !data.subscription
+        || !subscribers[data.subscription as WebSocketChannelName]
+      ) {
+        return;
+      }
+      // Call all subscribers for the channel
+      Object.values(subscribers[data.subscription as WebSocketChannelName]).forEach(
+        (subscriberCb) => subscriberCb(data.payload),
+      );
+    } catch (error) {
+      handleUnknownError(error);
+    }
+  }
+
+  /**
+   * Subscribes to websocket channels.
+   */
+  function subscribeToWebSocketChannels() {
+    wsClient.addEventListener('message', handleWebsocketMessage);
+
+    // Subscribe to all the channels
+    (Object.keys(channelsSubscribed) as WebSocketChannelName[]).forEach(
+      (channelName) => subscribeToWebSocketChannel(channelName),
+    );
+  }
+
+  /**
+   * UnSubscribes websocket channels.
+   */
+  function unsubscribeFromWebSocketChannels() {
+    // Remove event listener for messages
+    wsClient.removeEventListener('message', handleWebsocketMessage);
+
+    // UnSubscribe all channels
+    Object.keys(channelsSubscribed).forEach((channelName: string) => {
+      channelsSubscribed[channelName as WebSocketChannelName] = false;
+    });
+  }
+
+  /**
+   * This function creates a new WebSocket client and attempts to connect to the specified server.
+   * @param {string} url - The URL of the WebSocket server to connect to
+   */
+  async function connectToWebSocketServer(url: string) {
+    if (wsClient) {
+      unsubscribeFromWebSocketChannels();
+
+      if (isWsConnected.value) {
+        wsClient.close();
+      }
+    }
+
+    isWsConnected.value = false;
+
+    wsClient = new WebSocket(url);
+
+    wsClient.onopen = () => {
+      isWsConnected.value = true;
+      subscribeToWebSocketChannels();
+    };
+
+    wsClient.onclose = () => {
+      isWsConnected.value = false;
+    };
+  }
+
+  /**
+   * Subscribes to a specific websocket channel
+   * @param {WebSocketChannelName} channel - The websocket channel to subscribe to.
+   * @param {Function} [callback=(payload: any) => payload]
+   *    - The function to be called when a message is received on the specified channel.
+   * @returns {Function}
+   *    - A function that can be called to unsubscribe the specified callback from the channel.
+   */
+  function watchChannelUpdates(
+    channel: WebSocketChannelName,
+    // eslint-disable-next-line no-unused-vars
+    callback: (payload: any) => void,
+  ): () => void {
+    // subscribe to channel
```

Comment:
I think it's obvious what the below line does - I suggest to remove this comment.